genai_cookbook/nbs/1-introduction-to-agents.md

```diff
@@ -10,28 +16,30 @@ For example, suppose you are building a question-and-answer chatbot to help empl
 
 RAG addresses this issue by first retrieving relevant information from the company documents based on a user’s query, and then providing the retrieved information to the LLM as additional context. This allows the LLM to generate a more accurate response by drawing from the specific details found in the relevant documents. In essence, RAG enables the LLM to “consult” the retrieved information to formulate its answer.
 
-## Core components of a RAG application
+An agent with a retriever tool is one pattern for RAG, and has the advantage of deciding when to it needs to perform retrieval. This cookbook will describe how to build such an agent. 
+
+## Core components of a Agent application
 
-A RAG application is an example of a [compound AI system](https://bair.berkeley.edu/blog/2024/02/18/compound-ai-systems/): it expands on the language capabilities of the model alone by combining it with other tools and procedures.
+An agent application is an example of a [compound AI system](https://bair.berkeley.edu/blog/2024/02/18/compound-ai-systems/): it expands on the language capabilities of the model alone by combining it with other tools and procedures.
 
 When using a standalone LLM, a user submits a request, such as a question, to the LLM, and the LLM responds with an answer based solely on its training data.  
 
-In its most basic form, the following steps happen in a RAG application:
+In its most basic form, the following steps happen in an agent application with a retriever tool:
 
 1. **Retrieval:** The **user's request** is used to query some outside source of information. This might mean querying a vector store, conducting a keyword search over some text, or querying a SQL database. The goal of the retrieval step is to obtain **supporting data** that will help the LLM provide a useful response.
```

Comment:
This is different too in agents right? Maybe we can copy from the steps you wrote out below in `genai_cookbook/nbs/2-fundamentals-unstructured-chain.md`

Reply:
(ie the agent decides whether it needs to do retrieval, vs it being fixed as the first step)

Reply:
Or maybe we could actually have a table comparing RAG steps vs agent steps, so that the pic below still makes sense (we could mention that the diagram shows RAG but for agents, the LLM would be the one deciding to perform retrieval)

Reply:
lol oops. i inlined the steps from `genai_cookbook/nbs/2-fundamentals-unstructured-chain.md`

i think the image is still okay, i just added some text that says this is what happens if the retriever tool is selected.